[DDW-269] Automatic Daedalus updates

daniloprates opened the pull request:
This PR implements Daedalus app automatic update.

* [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1599222619006800)

## Todos

- [x] Newsfeed format
- [x] Edge cases:
  - [x] Update from the Newsfeed item is not valid
  - [x] Download fails
  - [x] Download success / corrupted file
  - [x] Download success / File is deleted
- [x] Copy/translation
- [x] Storybook stories
- [x] QA testing scenarios
- [x] ~E2E Tests~ (it will be handled by [DDW-353](https://jira.iohk.io/browse/DDW-353))

## Testing Checklist

* Builds: 
  * [`2.1.0`](https://buildkite.com/input-output-hk/daedalus/builds/14392)
  * [`2.2.0`](https://buildkite.com/input-output-hk/daedalus/builds/14320)
* [Scenarios](https://github.com/input-output-hk/daedalus/blob/03acf02a0ebbad7d870df364bc71ce1e891a743f/tests/app/e2e/features/app-update.feature)
* Testing
  * For scenarios with [2.2.0 not installed](https://github.com/input-output-hk/daedalus/blob/03acf02a0ebbad7d870df364bc71ce1e891a743f/tests/app/e2e/features/app-update.feature#L4), use [`2.1.0`](https://buildkite.com/input-output-hk/daedalus/builds/14392) build.
  * For scenario with [2.2.0 installed](https://github.com/input-output-hk/daedalus/blob/03acf02a0ebbad7d870df364bc71ce1e891a743f/tests/app/e2e/features/app-update.feature#L69), use [`2.1.0`](https://buildkite.com/input-output-hk/daedalus/builds/14392) build.
   * For [Quit and Install](https://github.com/input-output-hk/daedalus/blob/03acf02a0ebbad7d870df364bc71ce1e891a743f/tests/app/e2e/features/app-update.feature#L34) steps, it's not needed to actually install it, just close the installer.
  * To clean the state, before each scenario, open the [state directory](https://user-images.githubusercontent.com/1504716/92237801-5eb66580-ee8e-11ea-81c2-c0c737cfcec1.png) and delete the Downloads folder.
  * To [delete the temporary download file](https://github.com/input-output-hk/daedalus/blob/03acf02a0ebbad7d870df364bc71ce1e891a743f/tests/app/e2e/features/app-update.feature#L51), open the [state directory](https://user-images.githubusercontent.com/1504716/92237801-5eb66580-ee8e-11ea-81c2-c0c737cfcec1.png) and delete the file under the Downloads file.

* [Flowchart](https://user-images.githubusercontent.com/1504716/88346226-36c7d400-cd1e-11ea-963e-56845c1c59b1.png)

## QA Testing

[Testing sheet
](https://docs.google.com/spreadsheets/d/1mar7eIPxhbzl4lIz6mCZfzuUo8ip5lKDFuRTfY5TP1c/edit#gid=0)
### Build 14478:

**As expected:**
1 - The notification bell is visible
2 - The item is displayed on the Newsfeed first. It doesn't share the scrollbar with the rest of the items.
3 - The "Update available" overlay is displayed
4 - The "postpone" link works as expected. It is disabled/hidden when the “Close and update” button is clicked
5 - Update is successful on all platforms 
6 - The Information to update manually is displayed
7 - A progress bar is displayed on Linux while the update is performed.

**Fixed**
1 - The progress bar is displayed on Linux.

**Pending**
1 - The [logging](https://drive.google.com/file/d/15a4prBBzXyuXQgyiQJnNLTPteVwQV0xR/view) is still visible on Linux.

## Screenshots

![bell-1-loading](https://user-images.githubusercontent.com/1504716/91909882-65718c80-ec84-11ea-96ba-cc68b7723a1b.png)
![bell-2-home](https://user-images.githubusercontent.com/1504716/91909883-66a2b980-ec84-11ea-8bdc-034f7e407de8.png)
![en-1-newsfeed](https://user-images.githubusercontent.com/1504716/91909884-66a2b980-ec84-11ea-8107-333654f6635f.png)
![en-2-overlay-progress](https://user-images.githubusercontent.com/1504716/91909886-673b5000-ec84-11ea-9469-fe002777b29b.png)
![en-3-overlay-install](https://user-images.githubusercontent.com/1504716/91909889-67d3e680-ec84-11ea-8ded-1ca8d65e6077.png)
![en-4-overlay-install-manual-link](https://user-images.githubusercontent.com/1504716/91909890-67d3e680-ec84-11ea-9f4a-d11a6d32e757.png)
![en-5-error](https://user-images.githubusercontent.com/1504716/91909892-686c7d00-ec84-11ea-97fa-3b4ad4c0ba7c.png)
![jp-1-newsfeed](https://user-images.githubusercontent.com/1504716/91909896-686c7d00-ec84-11ea-93d0-4472406e21f3.png)
![jp-2-overlay-progress](https://user-images.githubusercontent.com/1504716/91909897-69051380-ec84-11ea-8a4e-e28d495b47aa.png)
![jp-3-overlay-install](https://user-images.githubusercontent.com/1504716/91909899-69051380-ec84-11ea-9b24-d70ee80c05df.png)
![jp-4-overlay-install-manual-link](https://user-images.githubusercontent.com/1504716/91910215-f7799500-ec84-11ea-9742-c7097926105d.png)
![jp-5-error](https://user-images.githubusercontent.com/1504716/91909904-699daa00-ec84-11ea-80cb-205336bf197e.png)


---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

nikolaglumac commented on package.json:
```diff
@@ -229,6 +229,7 @@
     "rust-cardano-crypto": "0.2.0",
     "safe-buffer": "5.2.0",
     "semver": "7.1.3",
+    "shasum": "^1.0.2",
```
@daniloprates please use fixed version here `"1.0.2"`

nikolaglumac commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "daedalus",
   "productName": "Daedalus",
-  "version": "2.2.0",
+  "version": "2.1.0",
```
@daniloprates this change is for testing purposes only right? We need to set it to 2.2.0 before merging 🙏

daniloprates replied:
Yes, we need it order to test the automatic download of `2.2.0`.

nikolaglumac reviewed (changes requested):
@daniloprates please check my comments 🙏

nikolaglumac commented:
@daniloprates looks like we also have some lint issues...

![Screenshot 2020-09-02 at 12 37 01](https://user-images.githubusercontent.com/376611/91971165-0b6ed680-ed19-11ea-904f-d11b0eb29175.png)

Please don't forget to run the translation manager 🙏

gnpf commented:
@daniloprates There are some missing translations in Japanese:

[Screenshot 1 - Download manually
](https://drive.google.com/file/d/1qN9HwtJJPXiW_MW2SD__CmgMbLI_I7bU/view?usp=sharing)[Screenshot 2 - MB Downloaded
](https://drive.google.com/file/d/1Rw7tcdOw1hoji-9khLaJV6gonQTHpH4m/view?usp=sharing)

Also, for the first screenshot, the link is not working (Only on JP).

gnpf commented:
@daniloprates In Linux, [the file in "Downloads"](https://drive.google.com/file/d/1RXkjpmbT4Xge4S2c33F4-yOSik0lLNwS/view?usp=sharing) is a **.exe**. 
Also, deleting that folder didn't work to clean the state on Mac or Lin.

daniloprates commented:
Thanks @gnpf. Fixed the wrong installers, will do the rest next week.

ManusMcCole commented:
Hi @daniloprates, I am seeing some strange behaviour on Mac. When I quit the installer mid download and then reopen Daedalus the download starts again instead of resuming from where it left off. Then once it finishes installing I run in to an error and the automatic update fails. I was able to reproduce this a few times where opening and closing Daedalus mid download caused this error. I am posting the video to the thread as it keeps coming back with an error here
<img width="1222" alt="Screenshot 2020-09-07 at 15 09 33" src="https://user-images.githubusercontent.com/56027085/92399321-0ac3af00-f122-11ea-901b-01b5361562c8.png">

ManusMcCole commented:
The same as above is also true for windows the error just looks a little different 
![image](https://user-images.githubusercontent.com/56027085/92401374-6d6a7a00-f125-11ea-90a2-2ae46725fdad.png)

gnpf commented:
@daniloprates Here is a list of new issues and some additional information for the previously reported:

`Note: Screenshots are linked`

1 - Regarding the issue on this comment, I found a clearer way to reproduce it:

> [Screenshot 2 - MB Downloaded ](https://drive.google.com/file/d/1Rw7tcdOw1hoji-9khLaJV6gonQTHpH4m/view?usp=sharing)

It happens when you switch from a language to another while downloading. If you open Daedalus in Japanese and switch to English, [you'll see the same inconsistency](https://drive.google.com/file/d/1brDSeGZwIZ9OpFqRihFeEPt0ru_iJfXW/view?usp=sharing). This does not reproduce [after you restart Daedalus.](https://drive.google.com/file/d/1jWs26S0UY741monnAxRtG8h4TpR7afik/view?usp=sharing)

2 - Sometimes, if you close Daedalus while downloading the update (close to finish), you'll see the "Update available" screen when [you reopen the application](https://drive.google.com/file/d/1tssttidUqHuE-mU9IWzo5KwTYCE6T92D/view?usp=sharing). In this case, the _**.crdownload**_ file is present on "Downloads".

3 - The "Update available" screen is still visible when [the folder "Downloads" is removed.](https://drive.google.com/file/d/1ENy0aciR7WzUFaRK5cnH-Z3hy1pLHj9Q/view?usp=sharing)

4 - On one occasion, the download was failing and I got [this screen](https://drive.google.com/file/d/1ATq43uUNOUIIfSLWejT2odWf2y7TkrVC/view?usp=sharing). Is this expected? I think a clearer message should be displayed if this happens. Also, the data on the progress bar was [unclear](https://drive.google.com/file/d/1U_w0MquUgQGNqUBFf3ZwaaqxGWotnbxZ/view?usp=sharing). 
The issue didn't reproduce after a restart.

a-rukin commented:
> 2 - The colors for the button in the white theme look inconsistent (Screenshot [**1**](https://drive.google.com/file/d/1SSEwognNYa9A_PfGPXTyyYD17V0B3zag/view?usp=sharing) & [**2**](https://drive.google.com/file/d/16kAxfSHbEnszVnAr99uUxHhgkxeTK2f7/view?usp=sharing)). Should the color of the buttons be the same as on "Rewards redemption" screen (Screenshot **[1](https://drive.google.com/file/d/14MGP-yueDMqD1yyhuZEpeg1FEtIWTd9R/view?usp=sharing)** & **[2](https://drive.google.com/file/d/17UzSrP10tUlpAqwVE1FxJiso-HVULjQP/view?usp=sharing)**)? Can you validate this @a-rukin?

Yes, colors of the buttons should be the same as on "Rewards redemption" screen

daniloprates commented:
@ManusMcCole Fixed issues when closing the app. [Build `2.1.0` is now #14365](https://buildkite.com/input-output-hk/daedalus/builds/14365) (still building).

nikolaglumac commented:
@daniloprates can you please fix this?

![scroller](https://user-images.githubusercontent.com/376611/92582373-2bf6dd80-f291-11ea-880f-8bbf4cbb9713.png)

nikolaglumac commented:
@daniloprates can we also automatically close the newsfeed sidebar once the update is downloaded and the update overlay is shown?

ManusMcCole commented:
Hi @daniloprates. On Shelley Testnet build on Windows the automatic update build is for Mainnet version of Daedalus. Usually it shows up with this error message. When it works occasionally it shows to be the incorrect update for Shelley testnet.
![Shelleygif (1)](https://user-images.githubusercontent.com/56027085/92586512-3b285c00-f28e-11ea-90ef-40114a546400.gif)

![image](https://user-images.githubusercontent.com/56027085/92586350-06b4a000-f28e-11ea-8aa4-15a12f3a4197.png)

nikolaglumac commented on source/renderer/app/domains/News.js:
```diff
@@ -170,6 +176,10 @@ class NewsCollection {
     return orderBy(read, 'date', 'asc');
   }
+
+  // @computed get update(): News | null {
```
@daniloprates should this be removed?

daniloprates replied:
[Yes](https://github.com/input-output-hk/daedalus/blob/da2a056e37b83cc33e94aa25a3d3faadea76917c/source/renderer/app/domains/News.js#L78)

nikolaglumac reviewed (changes requested):
@daniloprates please check my comments.

daniloprates commented on source/main/ipc/downloadManagerChannel.js:
```diff
@@ -131,29 +151,73 @@ const requestResumeDownload = async (
   );
 };
 
+// @APP_UPDATE TODO
```
@nikolaglumac testing dummy things

daniloprates commented on source/common/config/electron-store.config.js:
```diff
@@ -26,4 +26,6 @@ export const STORAGE_KEYS: {
   WALLET_MIGRATION_STATUS: 'WALLET-MIGRATION-STATUS',
   DOWNLOAD_MANAGER: 'DOWNLOAD-MANAGER',
+  APP_AUTOMATIC_UPDATE_FAILED: 'APP-AUTOMATIC-UPDATE-FAILED',
+  APP_UPDATE_COMPLETED: 'APP_UPDATE_COMPLETED',
```
✅

gnpf commented:
@daniloprates Here is an issue found on the latest 2.1.0 build:

- When the update file is already downloaded if you close and reopen Daedalus, [it tries to download it again](https://drive.google.com/file/d/1kWKpqPzwYKcVAQ-xcPTsEFIYtUUrGmCl/view?usp=sharing). Also, every time this happens, [a new _crdownload_ file is created on the _Downloads_ folder.](https://drive.google.com/file/d/1TxOtzk4y1Q2bdf6gKWxREB2anUm3HRuW/view?usp=sharing)

daniloprates commented:
> @daniloprates can you please fix this?
> 
> ![scroller](https://user-images.githubusercontent.com/376611/92582373-2bf6dd80-f291-11ea-880f-8bbf4cbb9713.png)

Done

daniloprates commented on source/renderer/app/containers/Root.js:
```diff
@@ -61,23 +62,27 @@ export default class Root extends Component<Props> {
       (isPageThatDoesntNeedWallets && !isNodeInStoppingSequence) ||
       (isProfilePage && (isNotEnoughDiskSpace || !isNodeInStoppingSequence))
     ) {
-      return <>{children}</>;
```
Yes, we're outside JSX so `<>{}</>` is useless

daniloprates commented:
> * Related to [the issue](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689469487) @ManusMcCole reported, the update file for all networks (Shelley T, Shelley QA, Flight, Staging, Testnet) [is always from Mainnet](https://drive.google.com/file/d/1nzmiLooMyY5fb1QTEfBDLh9hUKdZL97f/view?usp=sharing).

@gnpf @ManusMcCole this is expected, as the [update is a hardcoded](https://github.com/input-output-hk/daedalus/pull/2056/files#diff-0fdd5e05e2d1ce6b75204bee31ef8a7fR7-R19) for testing purposes.

Checking the other issues reported ⏳

daniloprates commented on source/renderer/app/components/widgets/NewsFeedIcon.scss:
```diff
@@ -2,9 +2,11 @@
-    background: var(--theme-news-feed-icon-dot-background-color);
+.dot {
+  &:after {
+    background: green;
```
> @daniloprates can we also automatically close the newsfeed sidebar once the update is downloaded and the update overlay is shown?

@nikolaglumac Done

daniloprates commented:
> @daniloprates can we also automatically close the newsfeed sidebar once the update is downloaded and the update overlay is shown?

@nikolaglumac Done

daniloprates commented on source/main/utils/downloadManager.js:
```diff
@@ -59,24 +59,24 @@ export const getOriginalFilename = ({
-  const { downloadId } = data;
-  await localStorage.setData(data, downloadId);
+  const { downloadId } = info;
+  await localStorage.setInfo(info, downloadId);
```
Nope. The download is divided into [Info & Data](https://github.com/input-output-hk/daedalus/blob/da2a056e37b83cc33e94aa25a3d3faadea76917c/source/common/types/downloadManager.types.js#L70-L91), **info** being the persistent information (filename, destination path, etc) and **data** the temporary bit (progress, time remaining, etc), so we need [different methods](https://github.com/input-output-hk/daedalus/blob/da2a056e37b83cc33e94aa25a3d3faadea76917c/source/main/utils/mainLocalStorage.js#L36-L62) for them.

daniloprates commented:
> When the update file is already downloaded if you close and reopen Daedalus, it tries to download it again. Also, every time this happens, a new crdownload file is created on the Downloads folder.. The original update file may be removed from the folder and this will still reproduce.

@gnpf @ManusMcCole fixed.

gnpf commented:
@daniloprates @ManusMcCole @nikolaglumac 
As this PR is getting a lot of comments, here is a summary of the issues reported so far and their status (Build 14379):

**Pending:**
1 - [Missing translations](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687286153). Also mentioned in [this comment (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742) 
2 - [Inconsistent colors for the white theme (# 2)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687293036)
3 - ["Update available" screen still visible when "Downloads" is removed.](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742 (# 3)) (Not sure if this is an issue)
4 - [Unclear screen when the download fails ](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742)(# 4)  


**Fixed:**
1 - [Wrong update files for Lin and Win (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687293036) 
2 - [Download not resumed](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688377250).  There are more comments related to this from [Manus](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688387186) and [me](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742) (#2)
3 - [Newsfeed and download manager share the same scrollbar](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689450639)
4 - [Newsfeed bar is still expanded after the download finishes](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689452119)
5 - [The download starts again when the update file is already available (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689553807)

**Not issues:**
1 - [All networks download Mainnet build](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689469487). Also mentioned in [this comment (# 2)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689553807)

**New:**
1 - [There is a delay when closing the app now](https://drive.google.com/file/d/19cGlSIG53-v8X7PtKcigfmEBS_RX7oXR/view?usp=sharing), that didn't happen on previous builds. It's more noticeable on Mac and Win.

daniloprates commented:
> **Pending:**
> 1 - [Missing translations](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687286153). Also mentioned in [this comment (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742)
> 2 - [Inconsistent colors for the white theme (# 2)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687293036)
> 3 - ["Update available" screen still visible when "Downloads" is removed.](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742) (Not sure if this is an issue)
> 4 - [Unclear screen when the download fails ](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742)(# 4)
> 
> **New:**
> 1 - [There is a delay when closing the app now](https://drive.google.com/file/d/19cGlSIG53-v8X7PtKcigfmEBS_RX7oXR/view?usp=sharing), that didn't happen on previous builds. It's more noticeable on Mac and Win.


@gnpf @ManusMcCole @nikolaglumac I've fixed the issues above in the [Build #14383](https://buildkite.com/input-output-hk/daedalus/builds/14383).

nikolaglumac commented:
@daniloprates one suggestion - in the scope of checks you run initially on Daedalus startup we should probably check if the downloaded installer is still present and then if it is not we should not show the "Update is ready" overlay. Instead, we should restart the download process, as in the current setup if you manually delete the downloaded file before launching Daedalus you will still get the update is ready overlay to be shown.

ManusMcCole commented:
Hi @daniloprates. I think the "Quit Daedalus and start the installation" button should be a more filled in green colour like the "Add wallet" button for the white theme when it is enabled?.Both are enabled in the screenshots. What do you think @a-rukin ?
<img width="557" alt="Screenshot 2020-09-10 at 11 05 42" src="https://user-images.githubusercontent.com/56027085/92716059-3a5afd00-f356-11ea-96a6-1a2e8c88e00a.png">
<img width="543" alt="Screenshot 2020-09-10 at 11 05 53" src="https://user-images.githubusercontent.com/56027085/92716075-3f1fb100-f356-11ea-8003-74173db0c25a.png">

a-rukin commented:
> Hi @daniloprates. I think the "Quit Daedalus and start the installation" button should be a more filled in green colour like the "Add wallet" button for the white theme when it is enabled?.Both are enabled in the screenshots. What do you think @a-rukin ?
> <img alt="Screenshot 2020-09-10 at 11 05 42" width="557" src="https://user-images.githubusercontent.com/56027085/92716059-3a5afd00-f356-11ea-96a6-1a2e8c88e00a.png">
> <img alt="Screenshot 2020-09-10 at 11 05 53" width="543" src="https://user-images.githubusercontent.com/56027085/92716075-3f1fb100-f356-11ea-8003-74173db0c25a.png">

Bordered button is used for overlays (and on hover it looks like the bottom one) 

Solid is used in the main interface

ManusMcCole commented:
Hi @a-rukin and @daniloprates . Your right both are correct and filled in green on the hover  👍

ManusMcCole commented:
Hi @daniloprates. How should Daedalus react when the download is failed ? I deleted the download mid download to simulate this and it freezes at this point. It has a similar error that Gabriela mentioned before.On one occasion it froze at this point also from opening and closing Daedalus many times mid download

<img width="680" alt="Screenshot 2020-09-10 at 11 24 53" src="https://user-images.githubusercontent.com/56027085/92718705-15688900-f35a-11ea-89e1-bea21c22713d.png">
<img width="557" alt="Screenshot 2020-09-10 at 11 44 01" src="https://user-images.githubusercontent.com/56027085/92719515-25cd3380-f35b-11ea-84ce-ddb84d0c52f4.png">

gnpf commented:
@daniloprates Here is a new summary (build 14384):

**Pending:**
1 - ["Update available" screen still visible when "Downloads" is removed](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742 (# 3)). Nikola also mentioned this in[ his last comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-690041950).
2 - [Unclear screen when the download fails ](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742)(# 4). Manus also mentioned this in [his last comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-690150710).

**Reincident**
1 - I managed to reproduce this issue that was already fixed:
> 5 - [The download starts again when the update file is already available (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689553807)

[Here is a new screenshot](https://drive.google.com/file/d/1fQvu-IAOfBrQhlSiU8Bk3LpTHDGuQQ_o/view?usp=sharing). Sorry if the capture is unclear, it was hard to reproduce.
It doesn't happen all the time and the reproduction steps are not clear to me. Will update later if I gather more information.

**Fixed:**
1 - [Wrong update files for Lin and Win (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687293036) 
2 - [Download not resumed](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688377250).  There are more comments related to this from [Manus](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688387186) and [me](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742) (#2)
3 - [Newsfeed and download manager share the same scrollbar](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689450639)
4 - [Newsfeed bar is still expanded after the download finishes](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689452119)
5 - [Missing translations](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687286153). Also mentioned in [this comment (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742) 
6 - [Inconsistent colors for the white theme (# 2)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-687293036)
7 -  [There is a delay when closing the app now](https://drive.google.com/file/d/19cGlSIG53-v8X7PtKcigfmEBS_RX7oXR/view?usp=sharing), that didn't happen on previous builds. It's more noticeable on Mac and Win.

**Not issues:**
1 - [All networks download Mainnet build](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689469487). Also mentioned in [this comment (# 2)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689553807)

**New:**
1 - The [green dot on the bell](https://drive.google.com/file/d/1SSALA7PMFodwDrE6Rwp9g3YPBNQc05ma/view?usp=sharing) is different from the [green dot on the notification](https://drive.google.com/file/d/14oLvPe8CdT7utjL9K9LISniMuBz7duo0/view?usp=sharing). I didn't find the first one on Zeplin. Is this expected? @a-rukin can you validate which is correct?
2 - After opening Daedalus, [the red dot on the bell is visible for a split second before switching to green](https://drive.google.com/file/d/1U-2kxA_lkx02A5-vIHhcCm1jkxDIZY_a/view?usp=sharing) (some frames on the capture were slowed down to make it easier to spot). Can this be improved?

daniloprates commented:
> @daniloprates one suggestion - in the scope of checks you run initially on Daedalus startup we should probably check if the downloaded installer is still present and then if it is not we should not show the "Update is ready" overlay. Instead, we should restart the download process, as in the current setup if you manually delete the downloaded file before launching Daedalus you will still get the update is ready overlay to be shown.

> Hi @daniloprates. How should Daedalus react when the download is failed ? I deleted the download mid download to simulate this and it freezes at this point. It has a similar error that Gabriela mentioned before.On one occasion it froze at this point also from opening and closing Daedalus many times mid download. Perhaps we need a button to restart download

@nikolaglumac @ManusMcCole @gnpf I've fixed both cases above in the [build #14391](https://buildkite.com/input-output-hk/daedalus/builds/14391):

* When Daedalus loads, if there's an update ready to be installed, now there's a check to see if the file still exists.

* When the file is downloaded in the middle of the process, it will fail 5 seconds after it reaches 100%. This is because there's no native way to check if the file still exists and as this is an edge case, the checking will happen only if the progress reached 100% and no "END" event was received after 5 seconds.

For both cases I'm simply displaying the error overlay. 
@nikolaglumac I would not restart the download process, as it could lead to an error loop.
I like Manu's idea of a Restart download button. But, for now, I'd go with this more conservative version and we can improve it later.

daniloprates commented:
Thanks @gnpf. Fixes in the [build 14391](https://buildkite.com/input-output-hk/daedalus/builds/14391):

> **Pending:**
> 1 - ["Update available" screen still visible when "Downloads" is removed](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742). Nikola also mentioned this in[ his last comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-690041950).
> 2 - [Unclear screen when the download fails ](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-688404742)(# 4). Manus also mentioned this in [his last comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-690150710).

1 and 2 fixed

> **Reincident**
> 1 - I managed to reproduce this issue that was already fixed:
> 
> > 5 - [The download starts again when the update file is already available (# 1)](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-689553807)

Fixed

> **New:**
> 1 - The [green dot on the bell](https://drive.google.com/file/d/1SSALA7PMFodwDrE6Rwp9g3YPBNQc05ma/view?usp=sharing) is different from the [green dot on the notification](https://drive.google.com/file/d/14oLvPe8CdT7utjL9K9LISniMuBz7duo0/view?usp=sharing). I didn't find the first one on Zeplin. Is this expected? @a-rukin can you validate which is correct?
> 2 - After opening Daedalus, [the red dot on the bell is visible for a split second before switching to green](https://drive.google.com/file/d/1U-2kxA_lkx02A5-vIHhcCm1jkxDIZY_a/view?usp=sharing) (some frames on the capture were slowed down to make it easier to spot). Can this be improved?

1 - Not fixed yet
2 - Can't be fixed. For the update there are some checking before the green dot is displayed

gnpf commented:
@daniloprates Great work, all previous issues are fixed. Here a few new issues I found:

1 - The **X** button on [this screen](https://drive.google.com/file/d/1HUjZFUK9Dj2qyzNv0wUGznWAeDIZTrmt/view?usp=sharing) is not working. Should it be there?

2 - When you get the "Update available" screen, if you wait a few seconds, [you'll get the "Unable to launch" error](https://drive.google.com/file/d/1-ukquKgAxqfV5z2Iu3Gc2zW22T6jR_e3/view?usp=sharing). This only happens if Daedalus' language is Japanese. 

Here are some additional questions:

1 - Is the article for the instructions to manually update available in Japanese? The link is still not working.

2 - Is the automatic update process implemented on Linux? I saw the [thread](https://input-output-rnd.slack.com/archives/GJMEW8FV3/p1599655008011900) on DevOps, but so far, [the update file is opened and nothing else happens.](https://drive.google.com/file/d/1A7iIvccpg4hbgPo29dzEiz8f6Bavv_0m/view?usp=sharing)

daniloprates commented:
Thanks @gnpf!

> @daniloprates Great work, all previous issues are fixed. Here a few new issues I found:
> 
> 1 - The **X** button on [this screen](https://drive.google.com/file/d/1HUjZFUK9Dj2qyzNv0wUGznWAeDIZTrmt/view?usp=sharing) is not working. Should it be there?
> 
> 2 - When you get the "Update available" screen, if you wait a few seconds, [you'll get the "Unable to launch" error](https://drive.google.com/file/d/1-ukquKgAxqfV5z2Iu3Gc2zW22T6jR_e3/view?usp=sharing). This only happens if Daedalus' language is Japanese.

1 and 2 fixed in the [build 14393](https://buildkite.com/input-output-hk/daedalus/builds/14393).

> Here are some additional questions:
> 
> 1 - Is the article for the instructions to manually update available in Japanese? The link is still not working.
> 
> 2 - Is the automatic update process implemented on Linux? I saw the [thread](https://input-output-rnd.slack.com/archives/GJMEW8FV3/p1599655008011900) on DevOps, but so far, [the update file is opened and nothing else happens.](https://drive.google.com/file/d/1A7iIvccpg4hbgPo29dzEiz8f6Bavv_0m/view?usp=sharing)

1 - Missed that, checking now.
2 - Yes, I'm going to implement a different action/copy for linux, where we're going to open the installer directory instead of running the file directly.

gnpf commented:
@daniloprates Regarding 14395:

1 - I can still reproduce this issue:

> 2 - When you get the "Update available" screen, if you wait a few seconds, you'll get the "Unable to launch" error. This only happens if Daedalus' language is Japanese.

[Here is a new gif.](https://drive.google.com/file/d/1MRpQX3v71ex3-dkcvU3ZnS71-prb-FEk/view?usp=sharing)

2 - The link for the manual update now redirects to **//daedaluswallet.io/en/download/** or **//daedaluswallet.io/ja/download/**. Is this expected? at least in English, [there was an article for this before](https://iohk.zendesk.com/hc/en-us/articles/360023850634).

3 - The issue with the **X** button is fixed 👍 

4 - On Linux, the "_Downloads_" folder is displayed when you click on "_Quit Daedalus and start installation_" 👍

daniloprates commented:
> 
> 1 - I can still reproduce this issue:
> 
> > 2 - When you get the "Update available" screen, if you wait a few seconds, you'll get the "Unable to launch" error. This only happens if Daedalus' language is Japanese.
> 
> [Here is a new gif.](https://drive.google.com/file/d/1MRpQX3v71ex3-dkcvU3ZnS71-prb-FEk/view?usp=sharing)

@gnpf I couldn't reproduce it 🤔 I've added some errors logging in the [build 14396](https://buildkite.com/input-output-hk/daedalus/builds/14396) so we can try to capture it.

> 2 - The link for the manual update now redirects to **//daedaluswallet.io/en/download/** or **//daedaluswallet.io/ja/download/**. Is this expected? at least in English, [there was an article for this before](https://iohk.zendesk.com/hc/en-us/articles/360023850634).

Yes, that article is not much of a help, better to send the user directly to Daedalus website.

nikolaglumac commented:
@gnpf @ManusMcCole can you please post the list of open issues? 🙏

ManusMcCole commented:
Hi @nikolaglumac and @daniloprates. The issue "2 - When you get the "Update available" screen, if you wait a few seconds, you'll get the "Unable to launch" error. This only happens if Daedalus' language is Japanese" is still present. I found this line in the logs which i believe is related to it 

 {"at":"2020-09-11T11:01:14.842Z","env":"shelley_testnet:macOS:19.6.0","ns":["daedalus","v2.1.0","*shelley_testnet*"],"data":{"error":{"errno":-2,"syscall":"stat","code":"ENOENT","path":"/Users/manusmccole/Library/Application Support/Daedalus Shelley Testnet v6/Downloads/Unconfirmed-2020-09-11T110039.0205Z.crdownload"}},"app":["daedalus"],"msg":"uncaughtException","pid":"","sev":"error","thread":""}

As for 4 - On Linux, the "Downloads" folder is displayed when you click on "Quit Daedalus and start installation" 👍 my Linux machine has been acting up. I made a new clone and am investigating now

ManusMcCole commented:
On Linux downloads folder is opening as expected 👍

ManusMcCole commented:
I do not see the x button for the Software update available screen on the newest builds @daniloprates 
<img width="1176" alt="Screenshot 2020-09-11 at 12 33 32" src="https://user-images.githubusercontent.com/56027085/92921017-38af3900-f42b-11ea-99cf-8e36977d8d63.png">

daniloprates commented:
@gnpf @ManusMcCole I've fixed the JP issue in the [build 14405](https://buildkite.com/input-output-hk/daedalus/builds/14405). There's still an error in the console that I couldn't handle, but the download and installation work well. Please, give it a go.

ManusMcCole commented:
Hi @daniloprates. There is a new issue now where the bell never turns green and the update available is never visible in the notification centre. The overlay just pops up once the download is complete.Also it still fails in Japanese 
<img width="794" alt="Screenshot 2020-09-11 at 12 47 52" src="https://user-images.githubusercontent.com/56027085/92922206-5a112480-f42d-11ea-8add-ba62579b5047.png">

gnpf commented:
@daniloprates @nikolaglumac Here is a summary of the latest issues:

1 - @ManusMcCole reported on [the previous comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691047815) that the download progress [is not visible on the notification bar.](https://drive.google.com/file/d/1QxzplH0-7iI1RI9N0zGig55Dz_xL3qKb/view?usp=sharing)

2 - Manus [also mentioned](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691047815) the update is still failing in Japanese. I also could reproduce this and got a console error, but only when [I click on the "Close and update" button](https://drive.google.com/file/d/18j_9zvKTaOwtB7oJo8JmABHp_K-zwVf9/view?usp=sharing).

3 - There a new link on ["update available" screen](https://drive.google.com/file/d/1PG-UtNqr17v_gt4tFcsYppHLzKfWCZvI/view?usp=sharing). Should it have the external link icon too?

4 - The link mentioned on the previous item, [is not visible in Japanese.](https://drive.google.com/file/d/1Qx_0YkGAvGbNtyALP47SGvsP40b6RI4J/view?usp=sharing)

5 - For Linux, the translation for the "Close and update" button in Japanese [is still missing](https://drive.google.com/file/d/1Qx_0YkGAvGbNtyALP47SGvsP40b6RI4J/view?usp=sharing) (I'm assuming the translation is pending)

daniloprates commented:
> @daniloprates @nikolaglumac Here is a summary of the latest issues:
> 
> 1 - @ManusMcCole reported on [the previous comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691047815) that the download progress [is not visible on the notification bar.](https://drive.google.com/file/d/1QxzplH0-7iI1RI9N0zGig55Dz_xL3qKb/view?usp=sharing)
> 
> 2 - Manus [also mentioned](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691047815) the update is still failing in Japanese. I also could reproduce this and got a console error, but only when [I click on the "Close and update" button](https://drive.google.com/file/d/18j_9zvKTaOwtB7oJo8JmABHp_K-zwVf9/view?usp=sharing).
> 
> 3 - There a new link on ["update available" screen](https://drive.google.com/file/d/1PG-UtNqr17v_gt4tFcsYppHLzKfWCZvI/view?usp=sharing). Should it have the external link icon too?
> 
> 4 - The link mentioned on the previous item, [is not visible in Japanese.](https://drive.google.com/file/d/1Qx_0YkGAvGbNtyALP47SGvsP40b6RI4J/view?usp=sharing)
> 
> 5 - For Linux, the translation for the "Close and update" button in Japanese [is still missing](https://drive.google.com/file/d/1Qx_0YkGAvGbNtyALP47SGvsP40b6RI4J/view?usp=sharing) (I'm assuming the translation is pending)

@gnpf @ManusMcCole @nikolaglumac 

* 1 and 2 fixed in the [build 14407](https://buildkite.com/input-output-hk/daedalus/builds/14407).
* 3: It's fake content to test the external link click. No need for Japanese.
* 4: [No copy/translation](https://input-output-rnd.slack.com/archives/CBECZ2K47/p1599774061001300) from Darko & Junko yet/

nikolaglumac commented:
@gnpf @ManusMcCole please test the fixes @daniloprates made and report new status - e.g. the list of remaining issues 🙏

ManusMcCole commented:
Hi @daniloprates..There is an error with the builds on Linux. Everytime as soon as Daedalus is opened it goes directly to the download overlay screen and the error "we were unable to launch the update installer automatically. Please manually update Daedalus to its newest version" is displayed. This Build working 100% on Mac and Windows
<img width="1152" alt="Screenshot 2020-09-14 at 10 11 17" src="https://user-images.githubusercontent.com/56027085/93067598-52db5800-f673-11ea-9617-4b8b8f980b9a.png">

ManusMcCole commented:
I tried to reproduce the above error on another Linux VM but the download was succesfull this time. Investigating further now with @mioriohk

ManusMcCole commented:
The Issue on Linux above is reproducible around 80% of the time

mioriohk commented:
> I tried to reproduce the above error on another Linux VM but the download was succesfull this time. Investigating further now with @mioriohk

@daniloprates , Hope that you are well :) I could replicate @ManusMcCole 's findings. However, the first time the error occurred after I selected the "I understand..." checkbox and clicked the "Quit Daedalus..." button. When I closed Daedalus and restarted, only the error was displayed. Please see gif.
![update](https://user-images.githubusercontent.com/67595081/93075282-7c00e600-f67d-11ea-864a-ce25948ed10e.gif)

daniloprates commented:
@ManusMcCole @mioriohk nice catch.

Some questions:
1. Does this happen before the green dot and download progress?
1. When clicking, besides the error message, is the directory opened or not?
1. Is there any error message in the dev console? Could you send me the logs?

Thanks.

ManusMcCole commented:
Hi @daniloprates. 1.For me this happens immediately when Daedalus is opened.2 It does not open the directory where the download should be. 3 No error in the dev console. Here are the logs 
[Logs.zip](https://github.com/input-output-hk/daedalus/files/5218559/Logs.zip)

daniloprates commented:
> 1.For me this happens immediately when Daedalus is opened.2 It does not open the directory where the download should be. 3 No er

Thanks @ManusMcCole. Could you try deleting the State Directory and try again, please?

ManusMcCole commented:
Hi @daniloprates. I tried again and i got the exact same results as above. I am attaching the logs again just incase they are different. @gnpf and @mioriohk can you please post if you experience different behaviour or any errors in the dev console 
[Logs.zip](https://github.com/input-output-hk/daedalus/files/5218678/Logs.zip)

gnpf commented:
@daniloprates @ManusMcCole @mioriohk 

I couldn't replicate [the mentioned issue](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691931623) in Ubuntu 20.04. I cleaned the environment and reinstalled multiple times and got [the expected results](https://drive.google.com/file/d/1dhwlJAIcN-2grlWXaasEBOIIiLFF7hL5/view?usp=sharing). 

The fixes mentioned in [this comment](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691722640) look good to me. I have some pending tests on different platforms, so I will add a detailed summary later.

daniloprates commented:
Thank you @ManusMcCole @gnpf @mioriohk. We will change the behavior for Linux, so I'll ignore this issue for now.

ManusMcCole commented:
Hi @daniloprates. the Downloads folder is not present on Linux when the overlay with the error message shows up immediately. I will resend you the logs now
<img width="1056" alt="Screenshot 2020-09-14 at 15 56 05" src="https://user-images.githubusercontent.com/56027085/93102450-6ef6ed80-f6a3-11ea-8237-1deb09519535.png">

ManusMcCole commented:
Here are the logs. There was an error with the file compression before hopefully it wont happen again
[Logs.zip](https://github.com/input-output-hk/daedalus/files/5219331/Logs.zip)

mioriohk commented:
> Hi @daniloprates. 1.For me this happens immediately when Daedalus is opened.2 It does not open the directory where the download should be. 3 No error in the dev console. Here are the logs
> [Logs.zip](https://github.com/input-output-hk/daedalus/files/5218559/Logs.zip)

As for me, I could see the Downloads folder in linux but I tested using Ubuntu 20.04 whereas Manus used 18.04

ManusMcCole commented:
Just a little to add to @mioriohk comment. In his scenario the downloads folder is present and Daedalus does not produce an error until after the checkbox is checked and the button "Quit Daedalus and start installation" is clicked. In my scenario the error shows up immediately upon Daedalus opening and the downloads folder is never present. I hope this helps to narrow down the issue @daniloprates

gnpf commented:
@daniloprates Here is a summary of issues and pending details on 14407:

1 - @ManusMcCole reported that the "Unable to launch" error is displayed every time [he opens Daedalus on Linux](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691931623). When checking the state directory, he found that [the "Downloads" folder was not present](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692116564). Logs are attached [here](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692123375). @mioriohk [could replicate this issue](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-691967228), but [I couldn’t](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692108077). Here are more details about this issue.

2 - I was able to reproduce [this issue again](https://drive.google.com/file/d/1pbhklPJ8VSENMDkRGH6ZAoctaA_JPdcC/view?usp=sharing):

> Sometimes, if you close Daedalus while downloading the update (close to finish), you'll see the "Update available" screen when you reopen the application. In this case, the .crdownload file is present on "Downloads".

It was hard to reproduce, so it doesn't happen every time. It's also worth mentioning, that I close the application multiple times while downloading.

3 - The translations for the "Quit and open directory" [are still pending](https://input-output-rnd.slack.com/archives/CBECZ2K47/p1599774061001300).

All the previous fixes seem to be working well, but I'll update this comment if something else comes up about this build.

daniloprates commented:
Thanks @gnpf @ManusMcCole @mioriohk I'm focusing on the Linux changes. Finished the Postpone button (needs different copy for Linux) and will finish the Linux installation tomorrow.

ManusMcCole commented:
Hi @daniloprates. Here is my findings for the newest build 14412. On linux the update download is not succesfull. The download progress bar is showing correctly and it completes as expected. However I am receiving the error "we were unable to launch the update installer automatically. Please manually update Daedalus to its newest version" after I click the checkbox and hit the "Quit Daedalus and start the installation" button. The downloads folder and download is present and the download works when I install it manually.Daedalus does not open the downloads folder as expected.Here are the logs
[Logs15-09.zip](https://github.com/input-output-hk/daedalus/files/5226373/Logs15-09.zip)

mioriohk commented:
> Hi @daniloprates. Here is my findings for the newest build 14412. On linux the update download is not succesfull. The download progress bar is showing correctly and it completes as expected. However I am receiving the error "we were unable to launch the update installer automatically. Please manually update Daedalus to its newest version" after I click the checkbox and hit the "Quit Daedalus and start the installation" button. The downloads folder and download is present and the download works when I install it manually.Daedalus does not open the downloads folder as expected.Here are the logs
> [Logs15-09.zip](https://github.com/input-output-hk/daedalus/files/5226373/Logs15-09.zip)

@daniloprates For build 14412, I observed the same behaviour as what @ManusMcCole explained above.

gnpf commented:
@daniloprates A few comments about the build 14412:

1 - I was able to reproduce [the issue](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692826605) mentioned by @ManusMcCole and @mioriohk [on Windows and macOS ](https://drive.google.com/file/d/1jb7-FlriiSY2Fccf8J-uevzNaOepVmG8/view?usp=sharing).

2 - Part of the text in Japanese [is now displayed in English for the "Progress" screen](https://drive.google.com/file/d/1mxLvSVe2C5W47iLZtjYvOUwip_Y5Jh2B/view?usp=sharing). Is this translation pending? Also reproducible on all platforms.

3 - The "Postpone" link works correctly across all platforms.

daniloprates commented:
@ManusMcCole @mioriohk @gnpf thanks guys. Is there any error in the dev console?

gnpf commented:
@daniloprates [Here is a screenshot](https://drive.google.com/file/d/1dbmX4qHwF9Y2VlhrFuhJysRNBxtJsUrF/view?usp=sharing) of the error on the console.

ManusMcCole commented:
Hi @daniloprates. Here are 2 error messages that appeared in the console 
<img width="691" alt="Screenshot 2020-09-15 at 21 20 48" src="https://user-images.githubusercontent.com/56027085/93261364-7ea52d80-f79a-11ea-820c-706861a01cb4.png">
<img width="568" alt="Screenshot 2020-09-15 at 21 28 06" src="https://user-images.githubusercontent.com/56027085/93261372-82d14b00-f79a-11ea-83de-ae21244d1f3d.png">

nikolaglumac commented on source/main/ipc/quitAppAndAppInstallUpdateChannel.js:
```diff
@@ -17,27 +19,56 @@ const quitAppAndAppInstallUpdateChannel: MainIpcChannel<
+        return false;
+      }
+
+      // For linux we execute the installer file
```
@daniloprates this is not true. We need to execute `update-runner filePath`...

gnpf commented:
@daniloprates Here's a summary for 14414:

When clicking the "Quit Daedalus and update" button we got various results:

1 - On Linux I got no error on the screen, but the update folder was not displayed. [A console error was displayed](https://files.slack.com/files-pri/T0N639Z4N-F01B00M8HV2/screenshot_from_2020-09-16_13-25-47.png). @ManusMcCole observed similar results.

2- @mioriohk got an error on the screen and no console error on [Linux](https://files.slack.com/files-pri/T0N639Z4N-F01AD4HKP5M/screenshot_2020-09-16_174814.gif) and [macOS](https://files.slack.com/files-pri/T0N639Z4N-F01AM5F4SQN/screenshot_2020-09-16_at_17.56.49.png)

3 - For Windows, @mioriohk got [no error on Daedalus](https://files.slack.com/files-pri/T0N639Z4N-F01ATTD4VS6/screenshot_2020-09-16_175827.gif) and [a system error while running the update file](https://files.slack.com/files-pri/T0N639Z4N-F01AD5MKABH/screenshot_2020-09-16_180019_2.gif).

Besides that, this issue is still reproducible:

> 2 - Part of the text in Japanese [is now displayed in English for the "Progress" screen](https://drive.google.com/file/d/1mxLvSVe2C5W47iLZtjYvOUwip_Y5Jh2B/view?usp=sharing). Is this translation pending? Also reproducible on all platforms.

ManusMcCole commented:
Hi @daniloprates.I know you said no need to test these builds but i was just having a look and i saw this new console error. Here is a <a href="https://drive.google.com/file/d/1lu2kQAHzwf2HOVAXf5i7aifNodNUGmj3/view?usp=sharing"><img style="width: 500px; max-width: 100%; height: auto" title="screenshot." /></a>. It appears after hitting the "Quit Daedalus and start the installation" button

ManusMcCole commented:
Hi @daniloprates. On the newest builds on Mac I am experiencing an error where attempting to close Daedalus causes the "Cardano node stopped" error and Daedalus freezes. It will not close no matter how many times i click on the quit button and have to resort to a Forced Quit in order to close. Here is a  <a href="https://drive.google.com/file/d/1wEGC2cew9VpHrbXVCWTyyoJr0folyPoh/view?usp=sharing">video</a>

gnpf commented:
@daniloprates Here's a summary for 14420 (More details on [this document](https://docs.google.com/spreadsheets/d/1mar7eIPxhbzl4lIz6mCZfzuUo8ip5lKDFuRTfY5TP1c/edit?usp=sharing)):

**As expected:**

1 - The notification bell is visible
2 - The item is displayed on the Newsfeed first. It doesn't share the scrollbar with the rest of the items.
3 - The "Update available" overlay is displayed
4 - The "postpone" link works as expected
5 - Update is successful on Windows and macOS

**Fixed:**

1 - Daedalus no longer freezes when attempting to close it. [This was reported ](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-694293491)by @ManusMcCole.

2 - The update file is executed correctly on macOS and Windows.

**Pending:**

1 - Missing translation in Japanese for the ["Download in progress" overlay](https://drive.google.com/file/d/1-ligm35z6LIPamaXYSZYobAfTustcjOX/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692858973).

**New:**

1 - The information for update manually is not displayed[ after the update file is deleted.](https://drive.google.com/file/d/1AjyQ6QMmV18_yY5T7ARWhTLN1oFLOE4K/view?usp=sharing)

gnpf commented:
@daniloprates Here's a summary for 14421 (More details on [this document](https://docs.google.com/spreadsheets/d/1mar7eIPxhbzl4lIz6mCZfzuUo8ip5lKDFuRTfY5TP1c/edit?usp=sharing)):

**As expected:**

1 - The notification bell is visible
2 - The item is displayed on the Newsfeed first. It doesn't share the scrollbar with the rest of the items.
3 - The "Update available" overlay is displayed
4 - The "postpone" link works as expected
5 - Update is successful on Windows and macOS
6 - The Information to update manually is displayed.

**Fixed:**

1 - The information to update manually is displayed on all platforms. This was reported [here](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-694425063).

**Pending:**

1 - Missing translation in Japanese for the ["Download in progress" overlay](https://drive.google.com/file/d/1-ligm35z6LIPamaXYSZYobAfTustcjOX/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692858973).

**New:**

1 - [An error is displayed on the console](https://drive.google.com/file/d/1vKgg10Gnz-KLv-NRO5w_o1E8kv85tui-/view?usp=sharing) when trying to update in Linux.

2 - The color of the spinner is incorrect for [Light Blue](https://drive.google.com/file/d/1feZkO0YXird1Xe-u02EzBQ4D-mE6xRNG/view?usp=sharing), [Cardano](https://drive.google.com/file/d/1Gb43z5FPm0fLCxu3j0TKHNDw-zQVZdQ0/view?usp=sharing) & [Dark Blue](https://drive.google.com/file/d/1jSjbj67pJhO-sistLklAW04u-GzLa1w9/view?usp=sharing).

gnpf commented:
@daniloprates Here's a summary for 14423 (More details on [this document](https://docs.google.com/spreadsheets/d/1mar7eIPxhbzl4lIz6mCZfzuUo8ip5lKDFuRTfY5TP1c/edit?usp=sharing)):

**As expected:**

1 - The notification bell is visible
2 - The item is displayed on the Newsfeed first. It doesn't share the scrollbar with the rest of the items.
3 - The "Update available" overlay is displayed
4 - The "postpone" link works as expected
5 - Update is successful on Windows and macOS
6 - The Information to update manually is displayed.

**Pending:**

1 - Missing translation in Japanese for the ["Download in progress" overlay](https://drive.google.com/file/d/1-ligm35z6LIPamaXYSZYobAfTustcjOX/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2056#issuecomment-692858973).
2 - The color of the spinner is incorrect for [Light Blue](https://drive.google.com/file/d/1feZkO0YXird1Xe-u02EzBQ4D-mE6xRNG/view?usp=sharing), [Cardano](https://drive.google.com/file/d/1Gb43z5FPm0fLCxu3j0TKHNDw-zQVZdQ0/view?usp=sharing) & [Dark Blue](https://drive.google.com/file/d/1jSjbj67pJhO-sistLklAW04u-GzLa1w9/view?usp=sharing).

**New:**

1 - The update file does not execute in Linux. There is no console error, and eventually, [Daedalus becomes unresponsive](https://drive.google.com/file/d/1uZ2gp23GkEu88ucQjb6fdNKjkybokc8a/view?usp=sharing).

**Suggestions:**

1 - I think the "Postpone" link could be disabled after the user clicks on "Close and update" button. At the moment you can click on it while Daedalus is closing/updating and it doesn't actually postpone the update, just closes the screen. Let me know what you think.

darko-mijic commented:
> 2 - Part of the text in Japanese is now displayed in English for the "Progress" screen. Is this translation pending? Also reproducible on all platforms.

This piece of copy is delivered through the newsfeed in both languages and it is unique for every update. We can use anything for testing. For example one of the newsfeed posts from the previous updates.

```
 "content": {
        "en-US": "This update significantly improves both blockchain synchronization speed and connection time, and fixes some network connection issues. Daedalus 2.2.0 also includes numerous enhancements and resolves some minor issues.",
        "ja-JP": "今回の更新ではまた、ブロックチェーンの同期スピードおよび接続時間が大幅に向上したほか、一部のネットワーク接続に関する不具合も修正されています。Daedalus 2.2.0には他にも多くの機能改善および一部の軽微な不具合の修正が含まれます。"
      }```

daniloprates commented:
@gnpf @mioriohk @ManusMcCole [build 14438](https://buildkite.com/input-output-hk/daedalus/builds/14438) fixes:

* Changing the app language would not change the update description
* The spinner color now follows the button text color for all themes
* The postpone link now behaves in the following ways, after clicking the main button:
  * Linux: The text changes to "Installing update...", no underline (not a link), no transparency
  * Other OS: The text remains as "Postpone the update", no underline (not a link), with transparency

* Linux installation is still failing.

gnpf commented:
@daniloprates Great job, most of the issues are fixed. I updated the description with the summary for 14438.
@ManusMcCole @mioriohk Feel free to update the summary if something else comes up on this build.

gnpf commented:
@daniloprates Great work! The update is now working on Linux. Here are a few unexpected behaviors I found on 14452:

1 - The update in Linux is working, but there are a lot of console errors. Here are a [screenshot](https://drive.google.com/file/d/15a4prBBzXyuXQgyiQJnNLTPteVwQV0xR/view?usp=sharing) and the [details](https://drive.google.com/file/d/1s_y49IxWbpD0vkKFA0hp_dppLUk9d7zZ/view?usp=sharing) for the errors I managed to capture.

2 - If the .crdownload file on "Downloads" is deleted and Daedalus reopen, the download resumes and completes, but a [new .crdownload is visible on the folder instead of the pkg, exe or bin](https://drive.google.com/file/d/1z39XF0wibcOEh4lQ35QHaywE8RXULbUA/view?usp=sharing). 

3 - As a continuation of the previous issue, if you click "update and restart" on Linux, [Daedalus updates](https://drive.google.com/file/d/1GjhW4r7OuR5yENpQlHT2nW6MfEFo2-hl/view?usp=sharing), even if the corresponding bin file is not present in "Downloads".

I'll wait until @mioriohk and @ManusMcCole test this to update the summary.

gnpf commented:
@daniloprates @cleverca22 The testing summary is updated on the description of this PR.

gnpf commented:
@daniloprates For 14467, there are a couple of issues with the download file:

1 - [The progress bar is displayed](https://drive.google.com/file/d/1xoy6XYUXlVWe2C3zXBjW1zSU63I05SA9/view?usp=sharing) even if the file is already downloaded.

2 - The update file [is downloaded everytime ](https://drive.google.com/file/d/12vOjLhOUC9ao3Ma4DAw90kvVF6zTrZOC/view?usp=sharing)Daedalus is open.

3 - The "Update available" screen is never displayed [after the bar reaches 100%](https://drive.google.com/file/d/18X2tUBcKm6k7x70ZUiiJkZtjwj1J_BpG/view?usp=sharing). Also a console error is visible.

These are partial results on Mac. I'll test on Win and Lin and update this if I found further issues.

gnpf commented:
@daniloprates @cleverca22
On build 14473:

1 - The issues with the update file are fixed.
2 - The logging [is still visible on Lin](https://drive.google.com/file/d/12F2N52A16oZjnTuBBC5PCfRiUwQWG7F0/view?usp=sharing).
3 - There is a progress bar for the installation on Linux, but[ it does not show progress.](https://drive.google.com/file/d/12F2N52A16oZjnTuBBC5PCfRiUwQWG7F0/view?usp=sharing) I'm not sure if this is still in progress.

ManusMcCole commented:
Hi @Danilo and @cleverca22 

Here is a summary of our findings for build 14475

1.The progress bar is working as expected. Although some of the themes colours are off. For <a href="https://drive.google.com/file/d/1uSO3YuJ_C1bcbVEk8VBV9qFm5tv3wPhZ/view?usp=sharing">Cardano</a>, <a href="https://drive.google.com/file/d/19rp4iTnHtQWjoZjs9Q0mZc15-zE-0GUg/view?usp=sharing">Dark Blue</a>, and <a href="https://drive.google.com/file/d/134JqbssZ6y-r3d5q6EbYCHVSmlLsUQDJ/view?usp=sharing">Light Blue</a> the colours are grey and white and I believe they should match this <a href="https://drive.google.com/file/d/134JqbssZ6y-r3d5q6EbYCHVSmlLsUQDJ/view?usp=sharing">Loading Bar Themes</a>

2.The update is working correctly on all OS

3.Logging messages are still appearing in the console on Linux as expected

a-rukin reviewed (approved):
👍

ManusMcCole reviewed (approved):
Great job @daniloprates 👍

gnpf reviewed (approved):
Great job @daniloprates and @cleverca22 🎉

mioriohk reviewed (approved):
Looks good to me 👍

nikolaglumac reviewed (approved):
Amazing work @daniloprates 🎉 
Proud of you 😄 💯
